libevmasm/RuleList.h

```diff
@@ -507,6 +507,18 @@ std::vector<SimplificationRule<Pattern>> simplificationRuleListPart7(
 		});
 	}
 
+	// Move AND with constant inside OR
+	rules.push_back({
+		// AND(OR(X, Y), A) -> OR(AND(X, A), AND(Y, A))
+		Builtins::AND(Builtins::OR(X, Y), A),
+		[=]() -> Pattern { return Builtins::OR(Builtins::AND(X, A), Builtins::AND(Y, A)); }
+	});
+	rules.push_back({
+		// AND(A, OR(X, Y)) -> OR(AND(X, A), AND(Y, A))
+		Builtins::AND(A, Builtins::OR(X, Y)),
+		[=]() -> Pattern { return Builtins::OR(Builtins::AND(X, A), Builtins::AND(Y, A)); }
```

Comment:
Could it be that
```suggestion
		[=]() -> Pattern { return Builtins::OR(Builtins::AND(A, X), Builtins::AND(A, Y)); }
```
would prevent those weird swaps from being generated in the cmdline test? Haven't really thought it though.

Reply:
I don't think this should matter. The CSE codegen should deal with that...

Reply:
Hm, yeah... I'm not sure where they are coming from...

Reply:
Also not sure it's a huge deal, though...